WordPress/src/main/java/org/wordpress/android/ui/prefs/MyProfileActivity.java

```diff
@@ -174,4 +175,17 @@ public void onEventMainThread(PrefsEvents.MyProfileDetailsChanged event) {
             refreshDetails();
         }
     }
+
+    public void onEventMainThread(PrefsEvents.MyProfileDataLoadSaveError event) {
+        if (!isFinishing()) {
+            if (event != null
```

Comment:
not really sure about this null check (while we need to check each nested element, it looks a bit awkward).
Also, not sure if we should take care of the 401 case in a specific manner; I'm doing it like so because that's the specific issue I'm addressing. More cases (and specific error texts) can be added if needed.